proxy/proxy/dial.go

```diff
@@ -56,9 +56,18 @@ func Dial(instance string) (net.Conn, error) {
 		dialClient.Unlock()
 	}
 
-	return c.Dial(instance)
+	return c.DialContext(ctx, instance)
+
+}
+
+// Dial does the same as DialContext but using context.Background() as the context.
+func Dial(instance string) (net.Conn, error) {
+	return DialContext(context.Background(), instance)
 }
 
+// ContextDialer is a convenience type to model the net.Dialer.DialContext() function
+type ContextDialer func(ctx context.Context, net, addr string) (net.Conn, error)
```

Comment:
Can the code just use the x/net/proxy.ContextDialer that already exists? I think that's whats happening, I can't tell where this type is being used actually.

Reply:
Oh right this code is completely unused. Too much copy-paste I guess. Removed in https://github.com/GoogleCloudPlatform/cloudsql-proxy/pull/483/commits/eec491fd6b9296a6f3d8d160bb16ab44a0e82a45